pkg/operator/controller/clientca-configmap/controller.go

```diff
@@ -0,0 +1,231 @@
+package clientcaconfigmap
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+	operatorcontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	"github.com/openshift/cluster-ingress-operator/pkg/util/slice"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName = "clientca_configmap_controller"
+)
+
+var log = logf.Logger.WithName(controllerName)
+
+// New creates a new controller that syncs client CA configmaps between the
+// config and operand namespaces.
+func New(mgr manager.Manager, config Config) (controller.Controller, error) {
+	reconciler := &reconciler{
+		cache:  mgr.GetCache(),
+		client: mgr.GetClient(),
+		config: config,
+	}
+	c, err := controller.New(controllerName, mgr, controller.Options{
+		Reconciler: reconciler,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	hasConfigMap := func(o client.Object) bool {
+		ic := o.(*operatorv1.IngressController)
+		return len(ic.Spec.ClientTLS.ClientCA.Name) != 0
+	}
+
+	// If the ingresscontroller's configmap reference changes, reconcile the
+	// ingresscontroller.
+	if err := c.Watch(
+		&source.Kind{Type: &operatorv1.IngressController{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				return hasConfigMap(e.Object)
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				return hasConfigMap(e.Object)
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				oldIC := e.ObjectOld.(*operatorv1.IngressController)
+				newIC := e.ObjectNew.(*operatorv1.IngressController)
+				oldName := oldIC.Spec.ClientTLS.ClientCA.Name
+				newName := newIC.Spec.ClientTLS.ClientCA.Name
+				if oldName != newName {
+					return true
+				}
+				if oldIC.DeletionTimestamp != newIC.DeletionTimestamp {
```

Comment:
nit: return the result of this expression.

Reply:
Done.